# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/udev/cleanup-id-logging.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- udev-182/src/ata_id/ata_id.c.vanilla	2024-11-25 21:02:30.446603719 +0100
+++ udev-182/src/ata_id/ata_id.c	2024-11-25 21:15:07.049924745 +0100
@@ -501,6 +501,7 @@
 
         fd = open(node, O_RDONLY|O_NONBLOCK);
         if (fd < 0) {
+                if (errno != ENOMEDIUM)
                 err(udev, "unable to open '%s'\n", node);
                 rc = 1;
                 goto exit;
--- udev-182/src/cdrom_id/cdrom_id.c.vanilla	2024-11-25 21:13:56.653259233 +0100
+++ udev-182/src/cdrom_id/cdrom_id.c	2024-11-25 21:16:06.063257126 +0100
@@ -939,8 +939,8 @@
                 nanosleep(&duration, NULL);
         }
         if (fd < 0) {
-                info(udev, "unable to open '%s'\n", node);
-                fprintf(stderr, "unable to open '%s'\n", node);
+                if (errno != ENOMEDIUM)
+                err(udev, "unable to open '%s'\n", node);
                 rc = 1;
                 goto exit;
         }
